checker/tests/nullness-enclosingexpr/NullnessEnclosingExprTest.java

```diff
@@ -0,0 +1,39 @@
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.initialization.qual.UnknownInitialization;
+
+class NullnessEnclosingExprTest {
+    class InnerWithImplicitEnclosingExpression {
+        // This will lead to a NPE, since NullnessEnclosingExprTest
+        // is not intialized yet.
+        InnerWithImplicitEnclosingExpression() {
+            NullnessEnclosingExprTest.this.f.hashCode();
+        }
+    }
+
+    class InnerWithUnknownInitializationEnclosingExpression {
+        InnerWithUnknownInitializationEnclosingExpression(
+                @UnknownInitialization NullnessEnclosingExprTest NullnessEnclosingExprTest.this) {
+            // TODO: This SHOULD lead to a NPE as we annotate @UnknownInitialization to the
```

Comment:
This seems like a separate false negative then. This PR checks the enclosing expression in the instantiation. Apparently, the annotation on the receiver here also isn't checked correctly. So open a new issue to fix that separate bug.
Then adapt the comment here: This should also never lead to an NPE, because that dereference should produce an type error. Add a link to the new issue and add a `// :: TODO error: (error.key)` for the error that you would expect here.

Reply:
Thanks! Done in my local.